uk.co.compucorp.civicrm.tasksassignments/api/v3/TASettings.php

```diff
@@ -1,70 +1,161 @@
+    'title' => 'Fields',
+    'description' => 'Which field(s) you want set',
+    'type' => CRM_Utils_Type::T_STRING,
+    'api.required' => 1,
+  ];
 }
 
+/**
+ * TASettings.set API
+ *
+ * Used to set a new value in the T&A Settings. Usage:
+ *
+ * civicrm_api3('TASettings', 'set', [
+ *   'field1' => 'bah',
+ *   'field2' => 'foo',
+ *   ...
+ * ])
+ *
+ * @param array $params
+ *
+ * @return array
+ * @throws \CiviCRM_API3_Exception
+ *   An Error will be thrown if any of the given fields doesn't exist
+ */
 function civicrm_api3_t_a_settings_set($params) {
-    
-    foreach ((array)$params['fields'] as $key => $value) {
-        if ($key === 'is_task_dashboard_default') {
-            if ($value == '1') {
-                CRM_Tasksassignments_DashboardSwitcher::switchToTasksAndAssignments();
-            } else {
-                CRM_Tasksassignments_DashboardSwitcher::switchToDefault();
-            }
-        }
-        if ($key === 'days_to_create_a_document_clone') {
-          $value = (int)$value;
-        }
-
-        $setting = civicrm_api3('OptionValue', 'get', array(
-            'option_group_id' => 'ta_settings',
-            'name' => $key,
-        ));
-
-        if (empty($setting['id'])) {
-            continue;
-        }
-
-        civicrm_api3('OptionValue', 'create', array(
-            'id' => $setting['id'],
-            'value' => $value,
-        ));
+  $validFields = _civicrm_api3_t_a_settings_valid_fields();
+
+  $requestFields = [];
+  if (!empty($params['fields']) && is_array($params['fields'])) {
+    $requestFields = $params['fields'];
+  }
+
+  foreach ($requestFields as $field => $value) {
+    if (!in_array($field, $validFields)) {
+      throw new \CiviCRM_API3_Exception("'{$field}' is not a valid field and cannot be set", 'invalid_field');
     }
-    
-    return civicrm_api3('TASettings', 'get', array(
-        'fields' => array_keys($params['fields']),
-        'sequential' => isset($params['sequential']) ? $params['sequential'] : 0,
-    ));
-}
\ No newline at end of file
+
+    if ($field === 'is_task_dashboard_default') {
+      if ($value == '1') {
+        CRM_Tasksassignments_DashboardSwitcher::switchToTasksAndAssignments();
+      }
+      else {
+        CRM_Tasksassignments_DashboardSwitcher::switchToDefault();
+      }
+    }
+    if ($field === 'days_to_create_a_document_clone') {
+      $value = (int) $value;
+    }
+
+    $settings = _civicrm_api3_t_a_settings_get_settings();
+
+    $settings['ta_settings'][$field] = $value;
+
+    civicrm_api3('Setting', 'create', $settings);
+  }
+
+  return civicrm_api3('TASettings', 'get', [
+    'fields' => array_keys($params['fields']),
+    'sequential' => isset($params['sequential']) ? $params['sequential'] : 0,
+  ]);
+}
+
+/**
+ * Returns the T&A Settings
+ *
+ * @return array
+ * @throws \CiviCRM_API3_Exception
+ */
+function _civicrm_api3_t_a_settings_get_settings() {
+  $settings = array_shift(civicrm_api3('Setting', 'get', [
```

Comment:
Could just do a `return` from here

Reply:
I've updated this function a little bit, so I think that now this is not relevant anymore